streampipes-client-go/streampipes/api/data_lake_measure_api.go

```diff
@@ -0,0 +1,121 @@
+The specific interaction behavior is provided by the method bound to the DataLakeMeasureApi struct.
+Currently, only some GET and Delete methods have been implemented.
+*/
+type DataLakeMeasureApi struct {
+	config      config.StreamPipesClientConnectionConfig
+	httpRequest streampipes_http.HttpRequest
+}
+
+func NewDataLakeMeasureApi(clientConfig config.StreamPipesClientConnectionConfig) *DataLakeMeasureApi {
+	//NewDataLakeMeasureApi is used to return an instance of *DataLakeMeasureApi,
+
+	return &DataLakeMeasureApi{
+		config:      clientConfig,
+		httpRequest: nil,
+	}
+}
+
+func (d *DataLakeMeasureApi) All() []data_lake.DataLakeMeasure {
+	//Get a list of all measurement series
+	//Deserializes the data into the corresponding DataLakeMeasure data model.
+
+	UnSerializer := serializer.NewBaseUnSerializer(serializer.WithUnSerializerDataLakeMeasures())
+	d.httpRequest = &streampipes_http.GetRequest{
+		HttpRequest:  streampipes_http.NewHttpRequest(d.config),
+		UnSerializer: UnSerializer,
+	}
+	d.ResourcePath(nil)
+	interfaces := d.httpRequest.ExecuteRequest(nil)
+	UnBaseSerializer := interfaces.(*serializer.UnBaseSerializer)
+	return *UnBaseSerializer.UnSerializerDataLakeMeasures
+}
+
+func (d *DataLakeMeasureApi) GetSingle(id string) data_lake.DataSeries {
+
+	//Get data from a single measurement series by a given id
+
+	UnSerializer := serializer.NewBaseUnSerializer(serializer.WithUnSerializerDataSeries())
+	d.httpRequest = &streampipes_http.GetRequest{
+		HttpRequest:  streampipes_http.NewHttpRequest(d.config),
+		UnSerializer: UnSerializer,
+	}
+	d.ResourcePath([]string{id})
+	interfaces := d.httpRequest.ExecuteRequest(nil)
+	UnBaseSerializer := interfaces.(*serializer.UnBaseSerializer)
+	return *UnBaseSerializer.UnSerializerDataLakeSeries
+}
+
+func (d *DataLakeMeasureApi) DeleteMeasurementInternalData(elementId string) string {
+	//Remove data from a single measurement series with given id
+
+	d.httpRequest = &streampipes_http.DeleteRequest{
+		HttpRequest:  streampipes_http.NewHttpRequest(d.config),
+		UnSerializer: nil,
+	}
+	d.ResourcePath([]string{elementId})
+	interfaces := d.httpRequest.ExecuteRequest(nil)
+	return interfaces.(string)
+}
+
+func (d *DataLakeMeasureApi) DeleteMeasurementSeries(elementId string) string {
+	//Drop a single measurement series with given id from Data Lake and remove related event property
+
+	d.httpRequest = &streampipes_http.DeleteRequest{
+		HttpRequest:  streampipes_http.NewHttpRequest(d.config),
+		UnSerializer: nil,
+	}
+	d.ResourcePath([]string{elementId, "drop"})
+	interfaces := d.httpRequest.ExecuteRequest(nil)
+	return interfaces.(string)
+}
+
+func (d *DataLakeMeasureApi) Create(element data_lake.DataLakeMeasure) error {
+
+	return fmt.Errorf("Not yet implemented")
+}
+
+func (d *DataLakeMeasureApi) Update(measure data_lake.DataLakeMeasure) error {
+	return fmt.Errorf("Not yet implemented")
+
+}
```

Comment:
We don't need to provide these unimplemented methods to the user. You could create an issue to track it. But we don't need to track it in the code.